src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs

```diff
@@ -0,0 +1,159 @@
+        {
+        }
+
+        /// <summary>
+        /// Generate URI from current buffer.
+        /// We can now use this url for ImageView.ResourceUrl
+        /// Note : the url lifecycle is same as ImageUrl and it's internal usage.
+        /// Store only ImageUrl.ToString() result and re-use that url is Undefined Behavior.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ImageUrl GenerateUrl()
+        {
+            mCachedImageUrl ??= new ImageUrl(Interop.EncodedImageBuffer.GenerateUrl(this.SwigCPtr.Handle), true);
+            return mCachedImageUrl;
+        }
+
+        /// <summary>
+        /// Get current raw buffer. (read-only)
+        /// Note : the raw buffer doesn't have memory ownership.
+        /// Access to write something to raw buffer is Undefined Behavior.
+        /// </summary>
+        /// This will not be public opened.
+        internal VectorUnsignedChar GetRawBuffer()
+        {
+            VectorUnsignedChar ret = new VectorUnsignedChar(Interop.EncodedImageBuffer.GetRawBuffer(this.SwigCPtr.Handle), false);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return ret;
+        }
+
+        /// <summary>
+        /// Dispose
+        /// </summary>
+        /// <param name="type"></param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override void Dispose(DisposeTypes type)
+        {
+            if (disposed)
+            {
+                return;
+            }
+
+            if (type == DisposeTypes.Explicit)
+            {
+                //Called by User
+                //Release your own managed resources here.
+                //You should release all of your own disposable objects here.
+                mCachedImageUrl?.Dispose();
+            }
+
+            base.Dispose(type);
+        }
+
+        /// This will not be public opened.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override void ReleaseSwigCPtr(HandleRef swigCPtr)
+        {
+            Interop.EncodedImageBuffer.DeleteEncodedImageBuffer(swigCPtr);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        /// <summary>
+        /// Get VectorUnsignedChar from System.IO.Stream.
+        /// This funcion is private (exist only for Constructor.)
+        /// </summary>
+        private static VectorUnsignedChar GetRawBuffrFromStreamHelper(System.IO.Stream stream)
+        {
+            if(stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+            if(!stream.CanRead)
+            {
+                throw new InvalidOperationException("stream don't support to read");
+            }
+            long streamLength = stream.Length;
```

Comment:
`Length` also not support all types of Stream
https://docs.microsoft.com/ko-kr/dotnet/api/system.io.stream.length?view=net-5.0#--

You do better to copy a stream on MemoryStream

https://github.com/Samsung/TizenFX/blob/90572287f263121d0e1bceaeb1273f99e8d5cd24/src/ElmSharp/ElmSharp/Image.cs#L484-L501

Reply:
If we change like that, Whole stream reading is the only possible works. Is it okay?

Reply:
I don't know exactly means of `Whole stream reading`. 
As I understand you need to know a stream size before create a native buffer.
So reading a whole stream is necessary to know a stream size.

Reply:
If we possible to append data into native buffer without a final size, we can use Stream without copy to memoryStream

Reply:
`VectorUnsignedChar` support `append` operation only for single byte.
(Appending raw-buffers are not support, but appending `VectorUnsignedChar` is support.)
That's why I get final size and copy it ;-;

Reply:
I imagine that some user push `used middle of stream` and try to load images from that.
But... I'll not care about them anymore.